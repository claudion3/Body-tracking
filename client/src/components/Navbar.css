.navbar {
	width: 100%;
	background: radial-gradient(
		circle,
		rgba(13, 8, 27, 1) 0%,
		rgba(26, 78, 62, 1) 98%
	);
	display: flex;
	justify-content: space-between;
	align-items: center;
	padding: 1.5rem;
	z-index: 50;
	position: sticky;
	top: 0;
}

.navbar_logo h2 {
	color: #f4f4f4;
	font-size: 1.4rem;
	cursor: pointer;
}

.cartLogo_badge {
	width: 30px;
	height: 30px;
	background: #f4f4f4;
	border-radius: 50%;
	margin-left: 8px;
	color: #171717;
	display: flex;
	justify-content: center;
	align-items: center;
	text-align: center;
	font-size: 1rem;
}

.hamburger_menu {
	width: 30px;
	height: 30px;
	display: none;
	flex-direction: column;
	justify-content: space-between;
	align-items: center;
	cursor: pointer;
}

.hamburger_menu > div {
	width: 100%;
	height: 3px;
	background: #f4f4f4;
}

.hamburger_menu:hover > div {
	background: #000301;
}

@media (max-width: 960px) {
	.hamburger_menu {
		display: flex;
	}
}

@media (max-width: 500px) {
	.navbar_logo h2 {
		font-size: 1rem;
	}
}
