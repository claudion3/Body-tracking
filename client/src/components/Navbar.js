import React from 'react';

import './Navbar.css';

const Navbar = ({ title }) => {
	return (
		<nav className='navbar' data-testid='navbar'>
			<div className='navbar_logo'>
				<h2>{title}</h2>
			</div>
			<div className='hamburger_menu'>
				<div></div>
				<div></div>
				<div></div>
			</div>
		</nav>
	);
};

export default Navbar;
