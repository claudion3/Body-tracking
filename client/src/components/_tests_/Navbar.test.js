import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar';

afterEach(() => {
	cleanup();
});

describe('Navbar component', () => {
	it('render without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Navbar></Navbar>, div);
	});

	it('renders Navbar correctly', () => {
		const { getByTestId } = render(<Navbar title='Truck App' />);
		expect(getByTestId('navbar')).toHaveTextContent('Truck App');
	});
	it('renders Navbar again correctly', () => {
		const { getByTestId } = render(<Navbar title='App log' />);
		expect(getByTestId('navbar')).toHaveTextContent('App log');
	});
	it('Matches snapshot', () => {
		const tree = renderer.create(<Navbar />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Matches snapshot with title=Truck App', () => {
		const tree = renderer.create(<Navbar title='Truck App' />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
