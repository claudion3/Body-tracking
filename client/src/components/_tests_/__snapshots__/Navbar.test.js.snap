// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Matches snapshot 1`] = `
<nav
  className="navbar"
  data-testid="navbar"
>
  <div
    className="navbar_logo"
  >
    <h2 />
  </div>
  <div
    className="hamburger_menu"
  >
    <div />
    <div />
    <div />
  </div>
</nav>
`;

exports[`Navbar component Matches snapshot with title=Truck App 1`] = `
<nav
  className="navbar"
  data-testid="navbar"
>
  <div
    className="navbar_logo"
  >
    <h2>
      Truck App
    </h2>
  </div>
  <div
    className="hamburger_menu"
  >
    <div />
    <div />
    <div />
  </div>
</nav>
`;
